lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+    my $ip    = $c->tx->remote_address;
+    return "$agent ($ip)";
+  };
+
+  # Use helper in action and template
+  get '/secret' => sub ($c) {
+    my $user = $c->whois;
+    $c->app->log->debug("Request from $user");
+  };
+
+  app->start;
+  __DATA__
+
+  @@ secret.html.ep
+  We know who you are <%= whois %>.
+
+A list of all built-in ones can be found in L<Mojolicious::Plugin::DefaultHelpers> and
+L<Mojolicious::Plugin::TagHelpers>.
+
+=head2 Plugins
+
+Plugins are application extensions that help with code sharing and organization. You can load a plugin with the keyword
+L<Mojolicious::Lite/"plugin">, which can omit the C<Mojolicious::Plugin::> part of the name, and optionally provide
+configuration for the plugin.
+
+  use Mojolicious::Lite;
+
+  plugin Config => {file => '/etc/myapp.conf', default => {foo => 'bar'}};
+
+  # Return configured foo value, or default if no configuration file
+  get '/foo' => sub ($c) {
+    my $foo = $c->app->config('foo');
+    $c->render(json => {foo => $foo});
+  };
+
+  app->start;
+
+L<Mojolicious::Plugin::Config> is a built-in plugin which can populate L<Mojolicious/"config"> using a config file.
+Plugins can also set up routes, hooks, handlers, or even load other plugins. A list of built-in plugins can be found at
+L<Mojolicious::Plugins/"PLUGINS">, and many more are available from
+L<CPAN|https://metacpan.org/search?q=Mojolicious+Plugin>.
+
+=head2 Placeholders
+
+Route placeholders allow capturing parts of a request path until a C</> or C<.> separator occurs, similar to the
+regular expression C<([^/.]+)>. Results are accessible via L<Mojolicious::Controller/"stash"> and
+L<Mojolicious::Controller/"param">.
+
+  use Mojolicious::Lite -signatures;
+
+  # /foo/test
+  # /foo/test123
+  get '/foo/:bar' => sub ($c) {
+    my $bar = $c->stash('bar');
+    $c->render(text => "Our :bar placeholder matched $bar");
+  };
+
+  # /testsomething/foo
+  # /test123something/foo
+  get '/<:bar>something/foo' => sub ($c) {
+    my $bar = $c->param('bar');
+    $c->render(text => "Our :bar placeholder matched $bar");
+  };
+
+  app->start;
+
+To separate them from the surrounding text, you can surround your placeholders with C<E<lt>> and C<E<gt>>, which also
+makes the colon prefix optional.
+
+=head2 Relaxed Placeholders
+
+Relaxed placeholders allow matching of everything until a C</> occurs, similar to the regular expression C<([^/]+)>.
+
+  use Mojolicious::Lite;
+
+  # /hello/test
+  # /hello/test.html
+  get '/hello/#you' => 'groovy';
+
+  app->start;
+  __DATA__
+
+  @@ groovy.html.ep
+  Your name is <%= $you %>.
+
+=head2 Wildcard placeholders
+
+Wildcard placeholders allow matching absolutely everything, including C</> and C<.>, similar to the regular expression
+C<(.+)>.
+
+  use Mojolicious::Lite;
+
+  # /hello/test
+  # /hello/test123
+  # /hello/test.123/test/123
+  get '/hello/*you' => 'groovy';
+
+  app->start;
+  __DATA__
+
+  @@ groovy.html.ep
+  Your name is <%= $you %>.
+
+=head2 HTTP methods
```

Comment:
This section seems out of place here, putting some placeholder sections above, and some below.